src/odemis/driver/orsay.py

```diff
@@ -1389,3 +1440,1097 @@ def terminate(self):
+        self.columnPressureConnector = None
+        self.compressedAirPressure = model.FloatContinuous(0, readonly=True, unit="Pa", range=COMP_AIR_PRESSURE_RNG)
+        self.compAirPressureConnector = None
+
+        self._connectorList = []
+
+        self.on_connect()
+
+    def on_connect(self):
+        """
+        Defines direct pointers to server components and connects parameter callbacks for the Orsay server.
+        Needs to be called after connection and reconnection to the server.
+        """
+
+        self._gunPump = self.parent.datamodel.HybridIonPumpGunFIB
+        self._columnPump = self.parent.datamodel.HybridIonPumpColumnFIB
+        self._interlockHVPS = self.parent.datamodel.HybridInterlockOutHVPS
+        self._interlockChamber = self.parent.datamodel.HybridInterlockInChamberVac
+        self._valve = self.parent.datamodel.HybridValveFIB
+
+        self._interlockHVPS.ErrorState.Subscribe(self._updateInterlockTriggered)
+        self._interlockChamber.ErrorState.Subscribe(self._updateInterlockTriggered)
+        self._valve.IsOpen.Subscribe(self._updateErrorState)
+        self._valve.IsOpen.Subscribe(self._updateValveOpen)
+        for device in self._devices_with_errorstates:
+            p = getattr(self.parent.datamodel, device).ErrorState
+            p.Subscribe(self._updateErrorState)
+
+        self.gunPumpOnConnector = OrsayParameterConnector(self.gunPumpOn, self._gunPump.IsOn)
+        self.columnPumpOnConnector = OrsayParameterConnector(self.columnPumpOn, self._columnPump.IsOn)
+        self.gunPressureConnector = OrsayParameterConnector(self.gunPressure, self._gunPump.Pressure)
+        self.columnPressureConnector = OrsayParameterConnector(self.columnPressure, self._columnPump.Pressure)
+        self.compAirPressureConnector = OrsayParameterConnector(self.compressedAirPressure,
+                                                                self.parent.datamodel.HybridGaugeCompressedAir.Pressure)
+
+        self._connectorList = [x for (x, _) in  # save only the names of the returned members
+                               inspect.getmembers(self,  # get all members of this FIB_source object
+                                                  lambda obj: type(obj) == OrsayParameterConnector  # get only the
+                                                  # OrsayParameterConnectors from all members of this FIB_source object
+                                                  )
+                               ]
+
+    def update_VAs(self):
+        """
+        Update the VA's. Should be called after reconnection to the server
+        """
+        self._updateErrorState()
+        self._updateInterlockTriggered()
+        self._updateValveOpen()
+        for obj_name in self._connectorList:
+            getattr(self, obj_name).update_VA()
+
+    def _updateErrorState(self, parameter=None, attributeName="Actual"):
```

Comment:
This is a thing from Orsay. Each of their parameters has an Actual and a Target attribute, which each contain a value. Generally when you want to access the value of a parameter, for example one called Pressure, you'd do that by using Pressure.Actual. But when subscribing to a parameter, the callback function needs to accept two arguments: the parameter (as an object) and the attribute name (as a string). This attribute name could then be "Actual" or "Target" (there are some other options, but those are irrelevant for now.) The main advantages of using a constant to store the "Actual" string would be that it only has to be changed in one place if Orsay decides to not call it Actual anymore, and that it decreases the risk of typos in the use of the "Actual" string. For the former argument, since the parameter.Actual notation is used whenever possible, there are many places this would have to be changed anyway (Ctrl+H will help). The latter argument does not weigh heavily enough I think, since any typo I would find through unittests. Do you agree?